routing/router.go

```diff
@@ -1658,7 +1658,18 @@ func (r *ChannelRouter) SendToRoute(hash lntypes.Hash, route *route.Route) (
 	// Since this is the first time this payment is being made, we pass nil
 	// for the existing attempt.
 	preimage, _, err := r.sendPayment(nil, payment, paySession)
-	return preimage, err
+	if err != nil {
+		// SendToRoute should return a structured error. In case the provided
+		// route fails, payment lifecycle will return a noRouteError with the
+		// structured error embedded.
+		if noRouteError, ok := err.(errNoRoute); ok {
+			return lntypes.Preimage{}, noRouteError.lastError
```

Comment:
hm, what happens if `lastError` is nil here?

Reply:
Should be possible to test by mocking a payment session that finds no route on first request.

Reply:
It should never happen, because SendToRoute always provides a route in the payment session. Added nil check to prevent crash.